tests/test_initialization.py

```diff
@@ -338,6 +338,194 @@ def test_lora_pissa_conversion_same_output_after_loading(self, data, tmp_path):
             model.linear.weight, model_converted.base_model.model.linear.base_layer.weight, atol=tol, rtol=tol
         )
 
+    def test_lora_pissa_conversion_same_output_after_loading_with_rank_pattern(self, data, tmp_path):
+        # same as above, but using rank_pattern
+        model = self.get_model()
+        output_base = model(data)[0]
+
+        # use rank_pattern here
+        config = LoraConfig(init_lora_weights="pissa", target_modules=["linear"], r=8, rank_pattern={"linear": 32})
+        peft_model = get_peft_model(deepcopy(model), config)
+        # save the initial model
+        peft_model.peft_config["default"].init_lora_weights = True
+        peft_model.save_pretrained(tmp_path / "init-model")
+        peft_model.peft_config["default"].init_lora_weights = "pissa"
```

Comment:
Help me unpack this bit of code.

We are using `pissa` for `init_lora_weights` already in the `LoraConfig`. So, I don't understand what would differ between `peft_model.peft_config["default"].init_lora_weights = True` and `peft_model.peft_config["default"].init_lora_weights = "pissa"`.

Reply:
That was due to a bug in original PiSSA implementation. I fixed it when I was implementing olora but I didn’t change tests. What would differ is `init_lora_weights `should be `True` in the config file. See [here](https://github.com/huggingface/peft/pull/1828#issuecomment-2160483919)